Framework/Core/include/Framework/InputRecord.h

```diff
@@ -385,7 +385,7 @@ class InputRecord
         // explicitely specify serialization method to ROOT-serialized because type T
         // is messageable and a different method would be deduced in DataRefUtils
         // return type with owning Deleter instance, forwarding to default_deleter
-        std::unique_ptr<ValueT const, Deleter<ValueT const>> result(DataRefUtils::as<ROOTSerialized<ValueT>>(ref).release());
+        std::unique_ptr<ValueT, Deleter<ValueT>> result(DataRefUtils::as<ROOTSerialized<ValueT>>(ref).release());
```

Comment:
What was the exact problem when using this option?

Reply:
ah, you need to do the same change in line 375 as well

Reply:
The return type had to be consistent within one branch, otherwise some tests (rightfully) would not compile. I would have to dump all the `const` in that branch, but this would confict with your intentions of avoiding copies for unserialized messages.

Reply:
yes true, there is also line 367 which you can not change ...

Reply:
Yes, that also spoke for decoupling messageable and non-messageable types here.